src/index.js

```diff
@@ -0,0 +1,26 @@
+export function recordPayloads(baseFetchQuery) {
+  const payloads = [];
+
+  function fetchQuery(...args) {
+    const i = payloads.length;
+    payloads.push(null);
+
+    return Promise.resolve(baseFetchQuery(...args)).then(payload => {
+      payloads[i] = payload;
+      return payload;
+    });
+  }
+
+  fetchQuery.toJSON = () => payloads;
+  return fetchQuery;
+}
+
+export function replayPayloads(baseFetchQuery, payloads) {
+  return function fetchQuery(...args) {
+    if (payloads.length) {
+      return payloads.shift();
```

Comment:
the assumption is that they are being replayed in the same order?

Reply:
yeah... i can/will switch to better keys if we end up merging/releasing this